### DISCUSSION TITLE
User settings for egs-parallel default parameters

### DISCUSSION THREAD

**Discussion by crcrewso** [nrc-cnrc/EGSnrc#679] [Ideas]
> Yes, that's a good idea. I won't change it for this merge, but will give it some thought for sure! I would prefer to gather all user settings in as few places as possible. Perhaps these egs-parallel defaults could go in `$EGS_CONFIG` (and even adjust the configuration script to define them when configuring EGSnrc)?

_Originally posted by @ftessier in https://github.com/nrc-cnrc/EGSnrc/pull/628#discussion_r601597149_

This is to track a feature that came up in discussions re the new egs-parallel  submission system. It would be nice if the default values were definable globally per user, so the user doesn't have to remember their flags.

**Comment by ftessier** [comment]
On the other hand, maybe it is better if users learn the flags 😆

**Reply by crcrewso** [reply]
I agree! It's entirely self serving. I support a few users on an internal cluster, and right now I have to manually reconfigure the scripts to prevent accidental submissions to `at` every time I update to the latest release. No matter how many times I remind them to put in all the tags, they do forget. I could satisfy this through maintaining my own branch but it would be nicer to just upstream the darned feature.

**Reply by ftessier** [reply]
Yep, makes sense, will do.

_As a stop gap the defaults in `egs-parallel` submit to the local cpu, so users will quickly feel the pain of simulations taking a long time to run! I find this works well to prevent job submission to the scheduler by mistake, i.e., you actually have to type options to submit outside the local computer. But I know it is a pain point for users to include the proper flags every time, and for admins to remind them. But I find it is useful too. It is like shell aliases: over time I have shied away from them (except for the widely adopted ones like `cp -i`; but even that is "dangerous" when you log on a computer without it!) so that I can get by as usual on any other system. I find that with `egs-parallel`, typing the flags all the time (well, CLI-completing or -searching them most of the time, really) helped me learn them quickly._

**Comment by ftessier** [comment]
I am moving this to the new Discussions tab, so as to keep Issues for actionable bugs.

**Comment by ftessier** [comment]
With Discussions we can also upvote ideas, to gauge what people would like to see implemented first!

**Comment by crcrewso** [comment]
Another Feature request 
egs-parallel-kill which would kill any open jobs, this will have to be written independently for each batch system, but I'm thinking it would at least be useful for the local scheduler at first

**Reply by ftessier** [reply]
Hmmm. Not sure about this one. I would rather not shadow the scheduler commands, just for the sake of having an `egs-parallel-*` named command. I mean unless `egs-parallel-kill` provides unique to EGSnrc itslef. Sorry for the thumbs down 😄 . On the other hand, I have an `egs-parallel-show` command (not added yet) which provides EGSnrc-specific information about running jobs.
